src/Java.Interop.Tools.Maven/Repositories/CachedMavenRepository.cs

```diff
@@ -0,0 +1,94 @@
+using System.Diagnostics.CodeAnalysis;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Java.Interop.Tools.Maven.Models;
+
+namespace Java.Interop.Tools.Maven.Repositories;
+
+/// <summary>
+/// Wraps an <see cref="IMavenRepository"/> and caches files in a local directory.
+/// </summary>
+public class CachedMavenRepository : IMavenRepository
```

Comment:
Naming question: I think this should be named `MavenRepositoryCache`.

Rationale/5 minutes of investigation: consider [dotnet/runtime](https://github.com/dotnet/runtime), commit dotnet/runtime@4765dd1b9f1aa58f16d6922438bcc6cb01b4a666:

```
# class contains `Cache`
% git grep 'public.*class.*Cache' src/libraries | wc -l 
      83

# class uses `Cached`
% git grep 'public.*class.*Caching' src/libraries | wc -l
      1

# class uses `Caching`
% git grep 'public.*class.*Caching' src/libraries | wc -l
      10

# class ends with `Cache`
% git grep 'public.*class.*Cache\>' src/libraries | wc -l
      20
```

Type names containing `Cache` include:

  * `CacheEntryExtensions`
  * `MemoryCacheEntryExtensions`

These don't feel like a good pattern to me, even if this has the most matches.

`Cached` has *one* match:

  * `XmlCachedSchemaSetResolver`

`Cache` as a suffix feels better:

  * `MemoryDistributedCache`
  * `RuleCache<T>`
  * `CredentialCache`

`Caching`, meanwhile, *only shows up as sub-namespaces*:

```
src/libraries/Microsoft.Extensions.Caching.Memory/ref/Microsoft.Extensions.Caching.Memory.cs:    public partial class MemoryDistributedCache : Microsoft.Extensions.Caching.Distributed.IDistributedCache
src/libraries/Microsoft.Extensions.Caching.Memory/ref/Microsoft.Extensions.Caching.Memory.cs:    public partial class MemoryCache : Microsoft.Extensions.Caching.Memory.IMemoryCache, System.IDisposable
src/libraries/Microsoft.Extensions.Caching.Memory/ref/Microsoft.Extensions.Caching.Memory.cs:    public partial class MemoryCacheOptions : Microsoft.Extensions.Options.IOptions<Microsoft.Extensions.Caching.Memory.MemoryCacheOptions>
src/libraries/Microsoft.Extensions.Caching.Memory/ref/Microsoft.Extensions.Caching.Memory.cs:    public partial class MemoryDistributedCacheOptions : Microsoft.Extensions.Caching.Memory.MemoryCacheOptions
src/libraries/Microsoft.Extensions.Diagnostics/tests/DefaultMetricsFactoryTests.cs:        public class NoCachingMeterFactory : IMeterFactory
src/libraries/System.Linq.Expressions/tests/Dynamic/CallSiteCachingTests.cs:    public class CallSiteCachingTests
src/libraries/System.Runtime.Caching/ref/System.Runtime.Caching.cs:    public abstract partial class CacheEntryChangeMonitor : System.Runtime.Caching.ChangeMonitor
src/libraries/System.Runtime.Caching/ref/System.Runtime.Caching.cs:    public abstract partial class FileChangeMonitor : System.Runtime.Caching.ChangeMonitor
src/libraries/System.Runtime.Caching/ref/System.Runtime.Caching.cs:    public sealed partial class HostFileChangeMonitor : System.Runtime.Caching.FileChangeMonitor
src/libraries/System.Runtime.Caching/ref/System.Runtime.Caching.cs:    public partial class MemoryCache : System.Runtime.Caching.ObjectCache, System.Collections.IEnumerable, System.IDisposable
```

So `Caching` feels like a bad idea.

Reply:
I think `MavenRepositoryCache` implies that this is a cache of `MavenRepository`, which it is not.

It is a wrapper around a `MavenRepository` which adds caching features to it.

It feels like `CachedMavenRepository` is better as `MavenRepository` is the noun and `Cached` (or `Caching`) is the adjective describing the modification to the `MavenRepository`.